support-workers/src/main/scala/com/gu/acquisitions/AcquisitionDataRowBuilder.scala

```diff
@@ -149,7 +152,12 @@ object AcquisitionDataRowBuilder {
         case (Collection, Saturday) => VoucherSaturday
         case (Collection, SaturdayPlus) => VoucherSaturdayPlus
         case (Collection, Sunday) => VoucherSunday
-        case _ => VoucherSundayPlus
+        case (Collection, SundayPlus) => VoucherSundayPlus
+        case (NoFulfilmentOptions, _) | (_, NoProductOptions) | (Domestic, _) | (RestOfWorld, _) |
+            (NationalDelivery, Saturday) | (NationalDelivery, Sunday) | (NationalDelivery, EverydayPlus) |
+            (NationalDelivery, SixdayPlus) | (NationalDelivery, WeekendPlus) | (NationalDelivery, SaturdayPlus) |
+            (NationalDelivery, SundayPlus) =>
+          InvalidPrintProduct(fulfilmentOptions, productOptions)
```

Comment:
good spot fixing this up, that could really skew things otherwise.
Regarding the exception idea, we would presumably get either a support workers failure or timeout in that case, whereas in this case we wouldn't know about it.  If we want this behaviour, is it worth sending a metric to cloudwatch and alarming on it?  I'm not sure either way but wanted to raise it.

Reply:
You’re right, this is too hidden. If the support-workers execution fails at this point (the `SendAcquisitionEvent` lambda within the parallel steps at the end), will it be straightforward to retry the lambda? I’ll test that now. I was worried that if it threw an exception it might be hard to recover the acquisition data to retry it, but I had forgotten this was within support-workers.

Reply:
Tested this: it’s quite straightforward to copy the input from the step function step and trigger a new execution with it. I think I’ll return to throwing a runtime exception in this case, and remove the `InvalidPrintProduct` type.